mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -1047,61 +1046,85 @@ pub async fn nexus_create(
 ) -> Result<(), Error> {
     // global variable defined in the nexus module
     let nexus_list = instances();
+
     if nexus_list.iter().any(|n| n.name == name) {
-        // instead of error we return Ok without making sure that also the
-        // children are the same, which seems wrong
+        // FIXME: Instead of error, we return Ok without checking
+        // that the children match, which seems wrong.
         return Ok(());
     }
 
-    let mut ni = Nexus::new(name, size, uuid, None);
+    // Create a new Nexus object, and immediately add it to the global list.
+    // This is necessary to ensure proper cleanup, as the code responsible for
+    // closing a child assumes that the nexus to which it belongs will appear
+    // in the global list of nexus instances. We must also ensure that the
+    // nexus instance gets removed from the global list if an error occurs.
+    nexus_list.push(Nexus::new(name, size, uuid, None));
```

Comment:
Could you get the nexus by index after you push it in directly, to avoid having to scan?

Reply:
Not sure if we can safely assume that the list cannot be updated mean-time. This seemed safer if not quite as efficient.
What I really wanted to do was to push the object on the list (a move) while simultaneously retaining a mutable reference to it, but alas the borrow checker says no.

Reply:
Yeah, makes sense!